extensions/gdata/src/com/google/refine/extension/gdata/SpreadsheetSerializer.java

```diff
@@ -0,0 +1,124 @@
+package com.google.refine.extension.gdata;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.model.AppendCellsRequest;
+import com.google.api.services.sheets.v4.model.BatchUpdateSpreadsheetRequest;
+import com.google.api.services.sheets.v4.model.BatchUpdateSpreadsheetResponse;
+import com.google.api.services.sheets.v4.model.ExtendedValue;
+import com.google.api.services.sheets.v4.model.Request;
+import com.google.api.services.sheets.v4.model.RowData;
+
+import com.google.refine.exporters.TabularSerializer;
+
+final class SpreadsheetSerializer implements TabularSerializer {
+    static final Logger logger = LoggerFactory.getLogger("SpreadsheetSerializer");
+    
+    private static final int BATCH_SIZE = 1000;
+    
+    private Sheets service;
+    private String spreadsheetId;
+    private List<Exception> exceptions;
+    
+    // A list of updates to apply to the spreadsheet.
+    private List<Request> requests = new ArrayList<>(); 
+    
+    private Request batchRequest = null;
+    private int row = 0;
+
+    private List<RowData> rows;
+    
+    SpreadsheetSerializer(Sheets service, String spreadsheetId, List<Exception> exceptions) {
+        this.service = service;
+        this.spreadsheetId = spreadsheetId;
+        this.exceptions = exceptions;
+    }
+    
+    @Override
+    public void startFile(JSONObject options) {
+    }
+    
+    
+    @Override
+    public void endFile() {
+        if (batchRequest != null) {
+            sendBatch(rows);
+        }
+        
+        BatchUpdateSpreadsheetRequest requestBody = new BatchUpdateSpreadsheetRequest();
+        requestBody.setIncludeSpreadsheetInResponse(false);
+        requestBody.setRequests(requests);
+
+        Sheets.Spreadsheets.BatchUpdate request;
+        try {
+            logger.info("spreadsheetId: " + spreadsheetId);
+            logger.info("requestBody:" + requestBody.toString());
```

Comment:
I assume we should remove this logging because the request body (and response below) could be quite verbose.

Reply:
I wouldn't remove it, it useful for debugging.  Instead just log this as DEBUG rather than INFO on slf4j

Reply:
will change to DEBUG. @wetneb I sent the info though the email to dev core. The token is to identify the application. Further more, before can use the API you need to use your own email to authorise.